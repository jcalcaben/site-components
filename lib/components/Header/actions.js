import React from "react";

import MoreVertIcon from "@material-ui/icons/MoreVert";
import SearchIcon from "@material-ui/icons/Search";

import defaultClasses from "./actions.module.css";

const Actions = (props) => {
  const { onSearchClick, onOverflowClick } = props;
  return (
    <section className={defaultClasses.root}>
      <button className={defaultClasses.searchButton} onClick={onSearchClick}>
        <SearchIcon
          fontSize={"large"}
          classes={{ root: defaultClasses.searchIcon }}
        />
      </button>
      <button className={defaultClasses.moreButton} onClick={onOverflowClick}>
        <MoreVertIcon
          fontSize={"large"}
          classes={{ root: defaultClasses.moreIcon }}
        />
      </button>
    </section>
  );
};

export default Actions;
