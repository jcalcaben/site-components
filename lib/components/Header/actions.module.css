.root {
  display: flex;
  width: auto;
  justify-self: right;
  padding: 0px;
  margin: 0px 3px;
}

.icon {
  vertical-align: middle;
}

.iconButton {
  background-color: var(--header-color);
  color: var(--secondary);
  border: none;
  padding: 0px;
  align-self: center;
  justify-self: right;
}

.moreButton {
  composes: iconButton;
}

.searchButton {
  composes: iconButton;
}

.moreIcon {
  composes: icon;
}

.searchIcon {
  composes: icon;
}
