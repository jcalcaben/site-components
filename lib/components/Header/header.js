import React from 'react'

import Actions from './actions'
import Navigation from './navigation'

import defaultClasses from './header.module.css'

const Header = props => {

    const { onNavigationClick, onOverflowClick, onSearchClick, children } = props;

    return (
        <header className={defaultClasses.root}>
            <Navigation clickAction={onNavigationClick} />
            {children}
            <Actions onOverflowClick={onOverflowClick} onSearchClick={onSearchClick} />
        </header>
    )

}

export default Header;