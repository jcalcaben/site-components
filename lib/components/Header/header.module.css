.root {
  background-color: var(--header-color);
  padding: 3px 2px;
  margin: 0px;
  height: 48px;
  display: grid;
  align-items: center;
  grid-template-columns: 1fr auto 1fr;
}
