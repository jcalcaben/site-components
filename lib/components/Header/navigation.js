import PropTypes from "prop-types";
import React from "react";

import MenuIcon from "@material-ui/icons/Menu";

import defaultClasses from "./navigation.module.css";

const Menu = (props) => {
  const { clickAction } = props;

  return (
    <button type="button" className={defaultClasses.root} onClick={clickAction}>
      <MenuIcon fontSize="large" classes={{ root: defaultClasses.icon }} />
    </button>
  );
};

Menu.propTypes = {
  clickAction: PropTypes.func,
};

export default Menu;
