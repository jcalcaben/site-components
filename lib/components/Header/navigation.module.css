.root {
  background-color: var(--header-color);
  color: var(--secondary);
  border: none;
  padding: 0px;
  align-self: center;
  justify-self: left;
}

.icon {
  vertical-align: middle;
}
