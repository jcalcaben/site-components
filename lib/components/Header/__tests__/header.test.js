import React from 'react'
import TestRenderer from 'react-test-renderer'

import Header from '../header'
import Navigation from '../navigation'
import Actions from '../actions'

const navClickFunction = jest.fn().mockName('Navigation click function');
const overflowClickFunction = jest.fn().mockName('Overflow click function');
const searchClickFunction = jest.fn().mockName('Search click function');

const baseProps = {
    onNavigationClick: navClickFunction,
    onOverflowClick: overflowClickFunction,
    onSearchClick: searchClickFunction
}

const Component = Header;

test('render header', () => {

    const testRenderer = TestRenderer.create(<Component {...baseProps} />)

    expect(testRenderer.toJSON()).toMatchSnapshot();

    expect(navClickFunction.mock.calls.length).toBe(0);
})

test('test navigation click', () => {

    const testRenderer = TestRenderer.create(<Component {...baseProps} />)

    const navigation = testRenderer.root.findByType(Navigation);
    navigation.findByType("button").props.onClick();

    expect(navClickFunction.mock.calls.length).toBe(1);
})

test('test search click', () => {

    const testRenderer = TestRenderer.create(<Component {...baseProps} />)

    const actions = testRenderer.root.findByType(Actions);

    actions.findAllByType("button")[0].props.onClick();

    expect(searchClickFunction.mock.calls.length).toBe(1);
})

test('test overflow click', () => {

    const testRenderer = TestRenderer.create(<Component {...baseProps} />)

    const actions = testRenderer.root.findByType(Actions);

    actions.findAllByType("button")[1].props.onClick();

    expect(overflowClickFunction.mock.calls.length).toBe(1);
})
