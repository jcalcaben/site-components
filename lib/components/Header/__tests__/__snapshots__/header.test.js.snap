// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render header 1`] = `
<header
  className="root"
>
  <button
    className="root"
    onClick={[MockFunction Navigation click function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root icon MuiSvgIcon-fontSizeLarge"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <section
    className="root"
  >
    <button
      className="searchButton"
      onClick={[MockFunction Search click function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root searchIcon MuiSvgIcon-fontSizeLarge"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </button>
    <button
      className="moreButton"
      onClick={[MockFunction Overflow click function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root moreIcon MuiSvgIcon-fontSizeLarge"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
    </button>
  </section>
</header>
`;
